chain/chain/src/resharding.rs

```diff
@@ -355,6 +361,11 @@ impl Chain {
         // Once we build the iterator, we break it into batches using the get_trie_update_batch function.
         let mut iter = iter;
         loop {
```

Comment:
Wow, I wonder how many other non-interrupt-able `loops` there are.

Reply:
Some that I know of are interruptible like cold store copy loop or the state sync dump. I think gc is limited to a const number of blocks per invocation. Dunno if there are more. I think block sync and catchup are interruptible too as I recall killing my node before it's synced.